### ISSUE TITLE
Random skipping in videos (not Sponsorblock)

### ISSUE THREAD

**Issue by opusforlife2** [TeamPiped/Piped#2517]
### Official Instance

- [X] The bug is reproducable on the [official hosted instance](http://piped.video/) or is API related.

### Describe the bug

Sometimes, videos skip a few seconds ahead (like 10s, 20s) for no apparent reason. Interestingly, trying to seek back using the arrow key doesn't work beyond the timestamp where the skip occurred. The video just resumes playing from that timestamp.

Edit: this is true if seeking to a timestamp within the skipped section. If seeking back to a point before that, the video plays, and then skips that section again.

The problem is easily solved by a simple page reload. The skipped section plays correctly.

These videos do not have any Sponsorblock segments in them.

### To Reproduce

1. Play a video.

The bug occurs randomly.

### Expected behavior

No unnecessary skipping.

### Logs/Errors

N/A

**Comment by Bnyro** [comment]
Are you sure it's not #2222, so the SponsorBlock markers don't show properly, and the skipping of SB segments just works properly?

**Comment by Fxsch** [comment]
I've experienced the same issue on some music videos

**Comment by Fxsch** [comment]
There's also LibreTube. It's an android app that uses piped and also lets you log in with a piped account. It doesn't have that issue tho, as far as I'm aware.

**Comment by NearestWater262** [comment]
Yes, LibreTube is what i was referring to, having this issue and also being very buggy @Fxsch

**Comment by FireMasterK** [comment]
Is there any way to reproduce this consistently? It would help to have a concise way to reproduce this issue, so it can be fixed.

**Comment by paricbat** [comment]
This issue happens to me sometimes and I can make it happen again by rewinding, maybe that could be somehow helpful.
It affects only a specific portion of the video, so my uneducated hunch would be that the player somehow receives corrupted parts of the video. This is because I noticed a similar thing happen in VLC when it couldn't read that part of the file, so it just skips it.

**Comment by cpiber** [comment]
Seems to be related to a bad gateway error:
![image](https://github.com/TeamPiped/Piped/assets/59023762/d259923b-56a0-4e0e-a070-6a297fe3b39b)

Not sure what causes it, since I don't have server logs (I don't run the instance), but resending the request succeeds, so it seems to be a temporary error. The problem can probably be solved by checking for the response code on the client side.

**Comment by rolsoft** [comment]
Just wanted to chime in to say that this has been happening for a long time on my instance.

It seems to be getting worse with time, the skipping happens mostly at the start of any video, and I have noticed it happens more often when set at a higher quality, so it probably is some kind of corrupted data received from piped-proxy or YT blocking the download of some parts of the video, idk. For me the only solution I have to watch YT videos now is to check my feed on my piped instance and then use yt-dlp to watch them downloaded, kinda annoying but it is the only thing that avoids the random skipping on videos.

I don't expect this issue to be solvable, because it doesn't seem to be piped's fault, probably the way the videos are downloaded is. 

Is there maybe a way to setup piped to be used with yt-dlp?, I know this would work for just small instances, because the download of each watched video would take a lot of disk space.
